modules/nlu/src/backend/middleware.ts

```diff
@@ -0,0 +1,89 @@
+import 'bluebird-global'
+import * as sdk from 'botpress/sdk'
+import util from 'util'
+
+import ScopedEngine from './engine'
+
+export const registerMiddleware = async (bp: typeof sdk, botScopedNlu: EngineByBot) => {
+  bp.events.registerMiddleware({
+    name: 'nlu.incoming',
+    direction: 'incoming',
+    handler: async (event: sdk.IO.Event, next: sdk.IO.MiddlewareNextCallback) => {
+      await processEvent(event)
+      next()
+    },
+    order: 10,
+    description:
+      'Process natural language in the form of text. Structured data with an action and parameters for that action is injected in the incoming message event.'
+  })
+
+  async function processEvent(event) {
+    if (['session_reset', 'bp_dialog_timeout', 'visit'].includes(event.type)) {
+      return
+    }
+
+    const botCtx = botScopedNlu[event.botId] as ScopedEngine
+
+    if (!botCtx) {
+      return
+    }
+
+    let eventIntent = { confidence: undefined, name: undefined }
+    let eventIntents = []
+
+    try {
+      const metadata = await botCtx.extract(event)
+
+      if (metadata) {
+        Object.assign(event, { nlu: metadata })
+        eventIntent = metadata.intent
+        eventIntents = metadata.intents
+
+        // if (botCtx.config.debugModeEnabled) {
```

Comment:
what's that comment ?

Reply:
Good catch. I meant to move that code somewhere else